garage/envs/mujoco/randomization/variation.py

```diff
@@ -170,17 +174,26 @@ def get_randomized_xml_model(self):
             # Check if the sampled value has the same shape with default value
             if np.array(c).shape != np.array(self._default_cache[v]).shape:
                 raise ValueError(
-                    "Sampled value you input %s don't match with default value %s in the xml node %s"
-                    % (c, self._default_cache[v], v.xpath))
+                    "Sampled value you input %s does not match with default "
+                    "value %s in the xml node %s" % (c, self._default_cache[v],
+                                                     v.xpath))
 
             if v.method == Method.COEFFICIENT:
-                e.attrib[v.attrib] = str(c * self._default_cache[v])
+                # store attrib. convert numpy print to xml-friendly print
+                e.attrib[v.attrib] = \
+                    str(c * self._default_cache[v]).strip("[],")
             elif v.method == Method.ABSOLUTE:
-                e.attrib[v.attrib] = str(c)
+                # store attrib. convert numpy print to xml-friendly print
+                e.attrib[v.attrib] = str(c).strip("[],")
             else:
                 raise ValueError("Unknown method")
 
-        return etree.tostring(self._parsed_model.getroot()).decode("ascii")
+        et = etree.ElementTree(self._parsed_model.getroot())
+        filename = os.path.join(MODEL_DIR, "varied_params.xml")
```

Comment:
do not write temporary files into the code. you have no idea where it is, and that location is not even guaranteed to be writable.

i think you want to use the tempfile library, though see my comment below about why writing files is problematic.